packages/dev/core/src/Compute/computeShader.ts

```diff
@@ -348,6 +348,20 @@ export class ComputeShader {
      * @returns True if the dispatch could be done, else false (meaning either the compute effect or at least one of the bound resources was not ready)
      */
     public dispatch(x: number, y?: number, z?: number): boolean {
+        return this._dispatchInternal({ x, y, z });
+    }
+
+    /**
+     * Dispatches (executes) the compute shader.
+     * @param buffer Buffer containing the number of workgroups to execute on the X, Y and Z dimensions
+     * @param offset Offset in the buffer where the workgroup counts are stored (default: 0)
+     * @returns True if the dispatch could be done, else false (meaning either the compute effect or at least one of the bound resources was not ready)
+     */
+    public dispatchIndirect(buffer: StorageBuffer | DataBuffer, offset: number = 0): boolean {
+        return this._dispatchInternal({ buffer, offset });
+    }
+
+    private _dispatchInternal(options: { x: number; y?: number; z?: number } | { buffer: StorageBuffer | DataBuffer; offset: number }): boolean {

```

Comment:
I wouldn't use objects to pass parameters, as this means that `dispatch` / `dispatchIndirect` will create new objects each time they are called, which can create a bit of GC pressure as dispatches can be made several hundred times per frame.

I would move the content of the `if (!this.fastmode) { ... }` condition to its own method (something like `_checkContext()`), and simply implement the dispatch methods as:
```typescript
public dispatch(x: number, y?: number, z?: number): boolean {
    if (!this.fastMode) {
        if (!this._checkContext()) {
            return false;
        }
    }

    this._engine.computeDispatch(this._effect, this._context, this._bindings, x, y, z, this._options.bindingsMapping, this.gpuTimeInFrame);

    return true;
}

public dispatchIndirect(buffer: StorageBuffer | DataBuffer, offset: number = 0): boolean {
    if (!this.fastMode) {
        if (!this._checkContext()) {
            return false;
        }
    }

    const buffer = ComputeShader._BufferIsDataBuffer(options.buffer) ? options.buffer : options.buffer.getBuffer();
    this._engine.computeDispatchIndirect(this._effect, this._context, this._bindings, buffer, offset, this._options.bindingsMapping, this.gpuTimeInFrame);

    return true;
}
```